src/test/kotlin/com/atlassian/performance/tools/hardware/HardwareApdexAnalysisIT.kt

```diff
@@ -0,0 +1,273 @@
+
+                //find the equivalent row form thw worst run
+                val worstRow = worst.metrics.asSequence()
+                    .groupBy { it.actionMetric.label }
+                    .asSequence().singleOrNull {
+                        it.key == entry.key
+                    }
+                if (worstRow != null) {
+                    tableAddActionRow(table, worst.run, worst.metrics.size, "", worstRow.value)
+                } else {
+                    table.addRowEmpty("?")
+                }
+
+                table.addRowEmpty()
+
+            }
+
+        println(table.generate())
+    }
+
+    private fun tableAddActionRow(table: GenericPlainTextTable, run: Int, allActionTotal: Int, actionLabel: String, metrics: List<ScoredActionMetric>) {
+
+        val actionTotal = metrics.size
+        val apdex = String.format("%.2f", Apdex().averageAllScoredMetrics(metrics))
+        val ok = metrics.filter { it.actionMetric.result == ActionResult.OK }.size
+        val error = metrics.filter { it.actionMetric.result == ActionResult.ERROR }.size
+        val errorPerc = String.format("%.2f", error * 100.0 / actionTotal)
+        val satisfactory = metrics.filter { it.score == 1.0f }.size
+        val satisfactoryPerc = String.format("%.2f", satisfactory * 100.0 / actionTotal)
+        val tolerable = metrics.filter { it.score == 0.5f }.size
+        val tolerablePerc = String.format("%.2f", tolerable * 100.0 / actionTotal)
+        val frustrated = metrics.filter { it.score == 0.0f && it.actionMetric.result == ActionResult.OK }.size
+        val profilePerc = String.format("%.2f", actionTotal * 100.0 / allActionTotal)
+
+        // double check our sums add up
+        if (satisfactory + tolerable + frustrated != ok) {
+            throw Exception("Calculated sum of SATISFACTORY ($satisfactory) + TOLERABLE ($tolerable) + FRUSTRATED ($frustrated) != calculated OK $ok")
+        }
+
+        if (ok + error != actionTotal) {
+            throw Exception("Calculated sum of OK ($ok) + ERROR ($error) != calculated TOTAL $actionTotal")
+        }
+
+        table.addRow(listOf(actionLabel, "$run", apdex, "$actionTotal", profilePerc, errorPerc, satisfactoryPerc, tolerablePerc))
+    }
+
+    private fun apdexMetrics(targetFile: File): ApdexResults {
+
+        val run = targetFile.name
+        val workspace = TestWorkspace(targetFile.toPath())
+
+        val cohortResults = targetFile.listFiles()
+            .filter { file ->
+                // I think this is a linear folder structure only ever 1 cohort per run? So potentially this could be simpler
+                val cohortFilter = "**/*${hardwareOfInterest.jira}*"
+                matchesSearchPattern(file.absolutePath, cohortFilter)
+            }
+            .mapNotNull { file ->
+                calculateApdexResults(workspace, file, run)
+            }
+            .first()
+
+        return cohortResults!!
```

Comment:
IntelliJ warns:
![image](https://user-images.githubusercontent.com/1194704/59277916-50fd3a80-8c61-11e9-999d-9af7935ae2ac.png)

Reply:
will tidy that up.

Reply:
Do you know if IntelliJ allows you to make the analysis warnings into build failures?

Reply:
It's a compiler-emitted warning
We can use this:
```
tasks.withType<KotlinCompile> {
    kotlinOptions.allWarningsAsErrors = true
}
```
Perhaps we'll finally fix the `kotlin-stdlib-jre7` deprecation warnings: https://ecosystem.atlassian.net/browse/JPERF-466